engine/kv_engine.cpp

```diff
@@ -1211,6 +1211,10 @@ Status KVEngine::Expire(const StringView str, TTLType ttl_time) {
         auto ret = res.entry_ptr->GetIndex().skiplist->SetExpireTime(
             expired_time, new_ts);
         res.s = ret.s;
+        {
+          std::unique_lock<std::mutex> skiplist_lock(skiplists_mu_);
+          outdated_skiplists_.emplace(res.entry_ptr->GetIndex().skiplist);
```

Comment:
expirable_skiplists_

Reply:
This is not safe. You should not modify elements in an std::set by SetExpireTime, as this will invalidate the ordering or elements.
Always erase the element, modify it, then insert it back to std::set.

Reply:
Not vital in this case but I agree